### PR TITLE
Subset excel

### LINKED ISSUE #5728: Add range argument when reading Excel files
I often find it awkward to just have the skip and lines to read with an Excel sheet.  An alternative in readxl is the **_range_** argument.  When it is given the **_rows to skip_** and **_maximum rows to import_**  are ignored, so they could be disabled, but that isn't essential.

We could have different formats in the dialogue.  I suggest a checkbox called Range.  If checked, then There is a **_From:_** with a default of **_A1_** and a **_To:_** with a default of the extent of the sheet if nothing were given.  (There should be space for 2 letters and quite a lot of numbers, e.g. AA1000000 I guess is the maximum?)

### PR DESCRIPTION
**Pull request by Ivanluv**
fixes #5728  @africanmathsinitiative/developers  this is ready for review

### CODE REVIEW

**Comment by Wycklife** [comment]
@Ivanluv Your branch has conflicts, please check

**Comment by maxwellfundi** [comment]
@Wycklife  I have now fixed the conflicts on this branch. You can now review @Ivanluv  work

**Comment by Wycklife** [comment]
@maxwellfundi  Fine

File: instat/dlgImportDataset.vb [hunk 1]
```diff
@@ -1,985 +1,1024 @@
...
+        ucrInputMissingValueStringExcel.SetRDefault(Chr(34) & "" & Chr(34))
+
+        ucrChkRange.SetText("Range:")
+        ucrChkRange.AddParameterPresentCondition(True, "range", True)
+        ucrChkRange.AddParameterPresentCondition(False, "range", False)
+        ucrChkRange.AddToLinkedControls({ucrInputTextFrom}, objValues:={True}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True, bNewLinkedChangeToDefaultState:=True, objNewDefaultState:="A1")
+        ucrChkRange.AddToLinkedControls({ucrInputTextTo}, objValues:={True}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True, bNewLinkedChangeToDefaultState:=True, objNewDefaultState:="AA100")
+
+        ucrInputTextFrom.SetParameter(New RParameter("from", bNewIncludeArgumentName:=False, iNewPosition:=0))
+        ucrInputTextTo.SetParameter(New RParameter("To", bNewIncludeArgumentName:=False, iNewPosition:=1))
```
Comment (lloyddewit) [inline instat/dlgImportDataset.vb:268]: Is there a special reason 'To' is capitalised (not consistent with 'from' parameter)?

**Comment by lloyddewit** [comment]
@Ivanluv I peer reviewed and it all looked very good. I only had one tiny comment.
Note: for some reason the Github diff for `dlgImportDataset.vb` did not work correctly, not sure why. I had to do the diff externally.

**Comment by Ivanluv** [comment]
@lloyddewit  it might be because the pull request on #5738  is done on the same dialogue

**Comment by lloyddewit** [comment]
@Ivanluv Yes, there was 3 weeks or so between implmentation and merge so another PR making significant changes to the same file could have confused the diff. No further comments from my side.

**Pull request merged**